Reachable unwrap panic in `sys_mmap()` caused by invalid `perms`

### Describe the bug
There is a out of bounds error in `sys_mmap()` at kernel/aster-nix/src/syscall/mmap.rs:27 when make a `mmap` syscall with specific arguments.

https://github.com/asterinas/asterinas/blob/0160a85ccd130c222ae9b8712b4102c726cba35c/kernel/aster-nix/src/syscall/mmap.rs#L27


### To Reproduce
1. Compile a program which calls `mmap`:
```C
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/mman.h>
#include <sys/wait.h>

int main(void)
{
  mmap(NULL, 0, 0xb, 0x4f142, -1, 0);
  return 0;
}
```
2. Run the compiled program in Asterinas.

### Expected behavior
Asterinas reports panic and is terminated.

### Environment
- Official docker asterinas/asterinas:0.6.2
- 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
- Asterinas version: main 0160a85c

### Logs
```
~ # /root/mmap_test.c
panicked at /root/asterinas/kernel/aster-nix/src/syscall/mmap.rs:27:61:
called `Option::unwrap()` on a `None` value
Printing stack trace:
   1: fn 0xffffffff8878fa40 - pc 0xffffffff8878fa58 / registers:

     rax               0x12; rdx 0xffffffff88a22ad8; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000015a9af0;

   2: fn 0xffffffff8878f820 - pc 0xffffffff8878f998 / registers:

     rax               0x12; rdx 0xffffffff88a22ad8; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000015a9b00;

   3: fn 0xffffffff88049000 - pc 0xffffffff8804900a / registers:

     rax               0x12; rdx 0xffffffff88a22ad8; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000015a9c80;

   4: fn 0xffffffff889a3040 - pc 0xffffffff889a30c2 / registers:

     rax               0x12; rdx 0xffffffff88a22ad8; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000015a9c90;

   5: fn 0xffffffff889a31e0 - pc 0xffffffff889a3220 / registers:

     rax               0x12; rdx 0xffffffff88a22ad8; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000015a9d20;

   6: fn 0xffffffff8898c7c0 - pc 0xffffffff8898c7d5 / registers:

     rax               0x12; rdx 0xffffffff88a22ad8; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000015a9d90;

   7: fn 0xffffffff88136280 - pc 0xffffffff88136315 / registers:

     rax               0x12; rdx 0xffffffff88a22ad8; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000015a9da0;

   8: fn 0xffffffff8839d1f0 - pc 0xffffffff883a1e5c / registers:

     rax               0x12; rdx 0xffffffff88a22ad8; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000015a9f30;

   9: fn 0xffffffff882b42e0 - pc 0xffffffff882b436e / registers:

     rax               0x12; rdx 0xffffffff88a22ad8; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000015c03d0;

  10: fn 0xffffffff88190fd0 - pc 0xffffffff88191b4f / registers:

     rax               0x12; rdx 0xffffffff88a22ad8; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000015c0570;

  11: fn 0xffffffff885cbd10 - pc 0xffffffff885cbd1e / registers:

     rax               0x12; rdx 0xffffffff88a22ad8; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000015c0f90;

  12: fn 0xffffffff887a5d90 - pc 0xffffffff887a5da6 / registers:

     rax               0x12; rdx 0xffffffff88a22ad8; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000015c0fb0;

  13: fn 0xffffffff887dd940 - pc 0xffffffff887dd9a9 / registers:

     rax               0x12; rdx 0xffffffff88a22ad8; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000015c0fd0;

  14: fn                0x0 - pc                0x0 / registers:

     rax               0x12; rdx 0xffffffff88a22ad8; rcx                0x1; rbx                0x0;
     rsi                0x0; rdi                0x0; rbp                0x0; rsp 0xffff8000015c1000;

[OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panicking.rs:107
(  2) /root/asterinas/ostd/src/panicking.rs:59
(  3) 2aghao2n2kcoquuybdyjuveav:?
(  4) ??:?
(  5) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:220
(  6) ??:?
(  7) 9yd76ggpeli67ls0owdwumxu5:?
(  8) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:220
(  9) /root/asterinas/kernel/aster-nix/src/syscall/mod.rs:328
( 10) /root/asterinas/kernel/aster-nix/src/thread/task.rs:69
( 11) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:79
( 12) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2077
( 13) /root/asterinas/ostd/src/task/task/mod.rs:310
make: *** [Makefile:153: run] Error 1
```

--- a/kernel/src/syscall/mmap.rs
+++ b/kernel/src/syscall/mmap.rs
@@ -24,7 +24,7 @@ pub fn sys_mmap(
     offset: u64,
     ctx: &Context,
 ) -> Result<SyscallReturn> {
-    let perms = VmPerms::from_posix_prot_bits(perms as u32).unwrap();
+    let perms = VmPerms::from_bits_truncate(perms as u32);
     let option = MMapOptions::try_from(flags as u32)?;
     let res = do_sys_mmap(
         addr as usize,

--- a/kernel/src/vm/perms.rs
+++ b/kernel/src/vm/perms.rs
@@ -16,12 +16,6 @@ bitflags! {
     }
 }
 
-impl VmPerms {
-    pub fn from_posix_prot_bits(bits: u32) -> Option<Self> {
-        VmPerms::from_bits(bits)
-    }
-}
-
 impl From<Rights> for VmPerms {
     fn from(rights: Rights) -> VmPerms {
         let mut vm_perm = VmPerms::empty();

